File: Data Structures & Algorithms/majority-element-ii/submission-0.cpp
class Solution {
public:
    vector<int> majorityElement(vector<int>& nums) {
        unordered_map<int, int> freq;

        for(auto num: nums) {
            freq[num]++;

            if(freq.size() > 2) {
                unordered_map<int, int> newFreq;

                for(auto x: freq) {
                    if(x.second > 1) {
                        newFreq[x.first] = x.second -1;
                    }
                }
                freq = newFreq;
            }
        }
        vector<int> ans;
        for(auto x: freq) {
            int cnt = 0;
            for(int num: nums) {
                if(num == x.first)
                    cnt++;
            }
            if(cnt > nums.size()/3){
                ans.push_back(x.first);
            }
        }
        return ans;
    }
};

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-0.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int ans = 1 ,l = 0;
        for(int i = 1;i<nums.length;i++) {
            if(nums[i] != nums[l]) {
                l++;
                // ans++;
                nums[l] = nums[i];
            }
        }
        return l+1;
    }
}